components/automate-gateway/pkg/authorizer/authorizer.go

```diff
@@ -4,113 +4,48 @@ import (
 	"context"
 	"fmt"
 
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-
-	"github.com/chef/automate/api/interservice/authz/common"
 	"github.com/chef/automate/components/automate-gateway/api/authz/pairs"
 	"github.com/chef/automate/components/automate-gateway/gateway/middleware"
 )
 
 type state struct {
-	v1, v2, next middleware.AuthorizationHandler
+	authHandler middleware.AuthorizationHandler
 }
 
-func NewAuthorizer(v1, v2 middleware.AuthorizationHandler) middleware.SwitchingAuthorizationHandler {
-	return &state{v1: v1, v2: v2, next: v1}
+func NewAuthorizer(authHandler middleware.AuthorizationHandler) middleware.SwitchingAuthorizationHandler {
+	return &state{authHandler: authHandler}
 }
 
 func (a *state) Handle(ctx context.Context,
 	subjects []string, projects []string, req interface{}) (context.Context, error) {
-
-	newCtx, err := a.next.Handle(ctx, subjects, projects, req)
-	st := status.Convert(err)
-	switch st.Code() {
-	case codes.OK:
-		return newCtx, nil
-	case codes.FailedPrecondition:
-		if a.fromStatus(st) {
-			return a.Handle(ctx, subjects, projects, req)
-		}
-		fallthrough
-	default: // any other error status
-		return ctx, err
-	}
+	return a.authHandler.Handle(ctx, subjects, projects, req)
```

Comment:
This is probably best left for another card but do we want to keep this Handler abstraction?

Reply:
No, should probably remove it... but that is part of the work referred to in my comment above about `SwitchingFilterHandler`.

Reply:
I'm confused - do you plan to do that in this PR or a follow-up?